omnigibson/systems/micro_particle_system.py

```diff
@@ -1042,21 +1131,35 @@ def generate_particle_instancer_on_object(
         else:
             prototype_indices = None
 
-        # Spawn the particles at the valid positions
-        cls.generate_particle_instancer(
-            n_particles=n_particles,
-            positions=particle_positions,
-            idn=idn,
-            particle_group=particle_group,
-            self_collision=self_collision,
-            prototype_indices=prototype_indices,
-        )
+        # Create a new particle instancer if a new idn is requested, otherwise use the pre-existing one
+        inst = cls.default_particle_instancer if instancer_idn is None else \
+            cls.particle_instancers.get(cls.particle_instancer_idn_to_name(idn=instancer_idn), None)
+
+        if inst is None:
+            inst = cls.generate_particle_instancer(
+                idn=instancer_idn,
+                particle_group=particle_group,
+                n_particles=n_particles,
+                positions=particle_positions,
+                self_collision=self_collision,
+                prototype_indices=prototype_indices,
+            )
+        else:
+            inst.add_particles(
+                positions=particle_positions,
+                prototype_indices=prototype_indices,
+            )
+
+        return inst
 
     @classmethod
     def remove_particle_instancer(cls, name):
         """
         Removes particle instancer with name @name from this system.
 
+        Note that if this is the default particle instancer (instancer 0), it will NOT delete this instancer,
+        but remove all of its owned particles
+
```

Comment:
Is your code actually doing that? Maybe it should raise an error if someone is trying to remove the default particle instancer? Also, what does `instancer.remove` do?

Reply:
conclusion: comment outdated

Reply:
removed!